### PR TITLE
Update to Beast 1.0.0-b34

### PR DESCRIPTION
**Pull request by vinniefalco**
Minor changes, should not do any harm. But probably won't fix anything in rippled either:

* Fix and tidy up CMake build scripts
* Require Visual Studio 2015 Update 3 or later
* Use fwrite return value in file_body
* Set internal state correctly when writing frames
* Add decorator unit test
* Add write_frames unit test
* Add io_service completion invariants test
* Update CMake scripts for finding packages
* Tidy up build settings
* Add missing dynabuf_readstream member

API Changes:

* Remove http Writer suspend and resume feature

WebSocket

* Move the handler, don't copy it

### CODE REVIEW

**Comment by vinniefalco** [comment]
*N.B.* Since this contains a git-subtree merge, it needs to be fast-forwarded onto **0.70.0-b4** and cannot be rebased.

File: src/beast/README.md [hunk 1]
```diff
@@ -102,19 +105,27 @@ Boost.Coroutine library. Please visit the Boost documentation for
 instructions on how to do this for your particular build system.
 
 For the examples and tests, Beast provides build scripts for Boost.Build (bjam)
-and CMake. Developers using Microsoft Visual Studio can generate Visual Studio
+and CMake. It is possible to generate Microsoft Visual Studio or Apple
+Developers using Microsoft Visual Studio can generate Visual Studio
```
Comment (seelabs) [inline src/beast/README.md]: Looks like this got garbled in editing.

**Comment by seelabs** [comment]
One minor issue in the docs, but no reason to hold up the PR. :+1:

File: src/beast/README.md (hunk 1 above)
Comment (vinniefalco) [inline src/beast/README.md]: You mean CMake can't generate Apple developers?!

Comment (vinniefalco) [inline src/beast/README.md]: Whoopsie, this has to be fixed in Beast first. I'll open an issue

**Comment by scottschurr** [comment]
@vinniefalco Shall I wait for your fix to the comments before making the next rippled release to develop?  I guess I'm wondering if you have an ETA for the fix to the comments...  Thanks.

**Comment by vinniefalco** [comment]
@scottschurr b34 was already merged to Beast's master branch, so I can't fix it in this pull request. You have two options, you could merge this as-is or wait for b35. If you wait for b35, then the subtree needs to be re-added. The documentation mixup is harmless to rippled, so there isn't great benefit to waiting. This pull request also updates rippled for some Beast API changes. So I think its better to merge this as-is.

**Comment by scottschurr** [comment]
@vinniefalco, thanks for the response.  I will merge as is.

**Pull request merged**